google-datacatalog-hive-connector/tests/google/datacatalog_connectors/hive/prepare/datacatalog_entry_factory_test.py

```diff
@@ -0,0 +1,256 @@
+
+    def test_make_entry_for_database_should_set_all_available_fields(
+            self):  # noqa:E125
+        databases = convert_json_to_metadata_object(
+            retrieve_json_file('databases_with_one_table.json'))['databases']
+
+        factory = datacatalog_entry_factory.DataCatalogEntryFactory(
+            DataCatalogEntryFactoryTestCase.__PROJECT_ID,
+            DataCatalogEntryFactoryTestCase.__LOCATION_ID,
+            DataCatalogEntryFactoryTestCase.__METADATA_SERVER_HOST,
+            DataCatalogEntryFactoryTestCase.__ENTRY_GROUP_ID)
+
+        entry_id, entry = \
+            factory.make_entries_for_database(databases[0])
+
+        self.assertEqual('default', entry_id)
+        self.assertEqual('default', entry.display_name)
+        self.assertEqual(
+            'projects/test_project/locations/location_id/'
+            'entryGroups/sql_database/entries/default', entry.name)
+        self.assertEqual('hive', entry.user_specified_system)
+        self.assertEqual('database', entry.user_specified_type)
+        self.assertEqual(
+            '//metadata_host//hdfs://'
+            'namenode:8020/user/hive/warehouse', entry.linked_resource)
+        self.assertIsNone(entry.source_system_timestamps.create_time)
+        self.assertIsNone(entry.source_system_timestamps.update_time)
+
+    def test_make_entry_for_table_should_set_all_available_fields(
+            self):  # noqa:E125
+        databases = convert_json_to_metadata_object(
+            retrieve_json_file('databases_with_one_table.json'))['databases']
+
+        table = databases[0].tables[0]
+
+        factory = datacatalog_entry_factory.DataCatalogEntryFactory(
+            DataCatalogEntryFactoryTestCase.__PROJECT_ID,
+            DataCatalogEntryFactoryTestCase.__LOCATION_ID,
+            DataCatalogEntryFactoryTestCase.__METADATA_SERVER_HOST,
+            DataCatalogEntryFactoryTestCase.__ENTRY_GROUP_ID)
+
+        entry_id, entry = \
+            factory.make_entry_for_table(table, 'my_datawarehouse')
+
+        self.assertEqual('my_datawarehouse__funds', entry_id)
+        self.assertEqual('funds', entry.display_name)
+        self.assertEqual(
+            'projects/test_project/locations/location_id/'
+            'entryGroups/sql_database/entries/'
+            'my_datawarehouse__funds', entry.name)
+        self.assertEqual('hive', entry.user_specified_system)
+        self.assertEqual('table', entry.user_specified_type)
+        self.assertEqual(
+            '//metadata_host//'
+            'hdfs://namenode:8020/user/hive/warehouse/funds',
+            entry.linked_resource)
+        self.assertEqual('2019-09-03 13:57:28+00:00',
+                         str(entry.source_system_timestamps.create_time))
+        self.assertEqual('2019-09-03 13:57:58+00:00',
+                         str(entry.source_system_timestamps.update_time))
+
+    def test_make_entry_for_table_invalid_database_name_should_set_all_available_fields(  # noqa:E501
```

Comment:
`test_make_entry_for_table_should_set_all_available_fields` already checks all available fields. A suggestion here, to keep the code more succinct, is to check only the fields that are different from the previous ones. Statements such as `self.assertEqual('funds', entry.display_name)` could be suppressed, then. WDYT? 🤔

Reply:
I prefer to keep it, the scenarios are different... if the assert logic was too complex then I'd agree, but what assurance do we have that both scenarios will go to the same code path?

What if the display name logic stops working only for the invalid assets name?

Since the assert is really simple, I'd like to keep it.

Reply:
I'm ok with either approach, thanks.